src/lib.rs

```diff
@@ -209,18 +209,26 @@ pub async fn init_redis_connection(
     let manager = RedisConnectionManager::new(format!("redis://{ip_addr}:{port}"))?;
     let pool = Pool::builder().build(manager).await?;
 
-    let mut id = CLIENT_ID.write().await;
-    *id = client_id;
-    let mut redis_pool = REDIS_POOL.write().await;
-    *redis_pool = Some(pool);
+    let mut pool_lock = REDIS_POOL
+        .try_write()
+        .map_err(|e| anyhow!("Redis pool error: {}", e))?;
+    *pool_lock = Some(pool);
+
+    let mut client_id_lock = CLIENT_ID
+        .try_write()
+        .map_err(|e| anyhow!("Redis client id error: {}", e))?;
+    *client_id_lock = client_id;
 
     Ok(())
 }
 
 #[allow(dead_code)]
 async fn log_to_redis(log: &str) -> anyhow::Result<()> {
-    let pool = REDIS_POOL.read().await;
-    let client_id = CLIENT_ID.read().await;
+    let pool_lock = REDIS_POOL.clone();
```

Comment:
```suggestion
async fn log_to_redis(log: &str) -> anyhow::Result<()> {
    let pool = REDIS_POOL.read().await;

    match &*pool {
        Some(pool) => {
            let mut conn = pool.get().await?;
            let client_id = CLIENT_ID.read().await;
            cmd("RPUSH")
                .arg(client_id.clone())
                .arg(log.to_owned())
                .query_async::<_, ()>(&mut *conn)
                .await
                .map_err(|e| anyhow!("Can't set log: {}", e))?;
        }
        None => return Err(anyhow!("Redis pool is not initialized")),
    }

    Ok(())
}
```

I'd suggest above. Reasons are
- Remove unnecessary clone()
- Remove unnecessary drop
- Call read() on CLIENT_ID only when the call is necessary.

I'd like to know what you think about this suggestion.

Reply:
I think those are valid changes and since clone() is pretty expensive in terms of memory I think these are changes are necessary since we would want to limit how much we use clone. But the only thing is I believe we need the ```let_ = ``` on line 234 since we would need to ignore the resulting value from the query.

Reply:
Done